src/test/java/com/google/cloud/compute/v1/it/ITInstanceTest.java

```diff
@@ -42,42 +76,191 @@ public static void setUp() throws IOException {
+    Instance instanceResource =
+        Instance.newBuilder()
+            .setName(INSTANCE)
+            .setMachineType(MACHINE_TYPE)
+            .addDisks(DISK)
+            .addNetworkInterfaces(NETWORK_INTERFACE)
+            .build();
+    instanceClient.insertInstance(PROJECT_ZONE_NAME.toString(), instanceResource);
   }
 
   @AfterClass
   public static void tearDown() {
+    instanceClient.deleteInstance(FORMATTED_INSTANCE);
     instanceClient.close();
   }
 
   @Test
-  public void testInsertInstance() {
-    String machineType =
-        ProjectZoneMachineTypeName.of("n1-standard-1", DEFAULT_PROJECT, ZONE).toString();
-    AttachedDisk disk =
-        AttachedDisk.newBuilder()
-            .setBoot(true)
-            .setAutoDelete(true)
-            .setType("PERSISTENT")
-            .setInitializeParams(
-                AttachedDiskInitializeParams.newBuilder().setSourceImage(DEFAULT_IMAGE).build())
-            .build();
-    Instance instance =
-        Instance.newBuilder()
-            .setName("mytestinstancetemplate")
-            .setMachineType(machineType)
-            .addDisks(disk)
+  public void getInstanceTest() {
+    assertInstanceDetails(getInstance());
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void listInstancesTest() {
+    ListInstancesHttpRequest instancesHttpRequest =
+        ListInstancesHttpRequest.newBuilder().setZone(PROJECT_ZONE_NAME.toString()).build();
+    for (Instance instance : instanceClient.listInstances(instancesHttpRequest).iterateAll()) {
+      if (instance.getName().equals(INSTANCE)) {
+        assertInstanceDetails(instance);
+      }
+    }
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void updateInstanceTest() {
+    List<String> requiredProperties =
+        Arrays.asList("name", "machineType", "disks", "networkInterfaces");
+    List<String> updatableProperties = Arrays.asList("metadata", "fingerprint", "description");
+    List<String> fieldMask =
+        Arrays.asList(
+            requiredProperties.get(0),
+            requiredProperties.get(1),
+            requiredProperties.get(2),
+            requiredProperties.get(3),
+            updatableProperties.get(0),
+            updatableProperties.get(1),
+            updatableProperties.get(2));
+
+    Instance instance = getInstance();
+    assertInstanceDetails(instance);
+
+    UpdateInstanceHttpRequest request =
+        UpdateInstanceHttpRequest.newBuilder()
+            .setMostDisruptiveAllowedAction("REFRESH")
+            .setMinimalAction(null)
+            .setInstance(FORMATTED_INSTANCE)
+            .setInstanceResource(
+                instance
+                    .toBuilder()
+                    .setDescription("updated description")
+                    .setFingerprint(instance.getFingerprint())
+                    .build())
+            .addAllFieldMask(fieldMask)
             .build();
+    if (instance.getStatus().equals("RUNNING")) {
```

Comment:
You need to change 
`if (instance.getStatus().equals("RUNNING"))`
->
`assertThat(instance.getStatus()).isEqualTo("RUNNING"))`
otherwise you test might silently pass without runing

Reply:
Thanks for suggestion, but this `assertThat(instance.getStatus()).isEqualTo("RUNNING"))` will fail in case of instance in `STAGGING `mode.

so better to wait until instance get `RUNNING`.